modules/secrets-store-aws-secrets-manager.adoc

```diff
@@ -0,0 +1,357 @@
++
+.Example `secret-provider-class-aws.yaml`
+[source,yaml]
+----
+apiVersion: secrets-store.csi.x-k8s.io/v1
+kind: SecretProviderClass
+metadata:
+  name: my-aws-provider                   <1>
+  namespace: my-namespace                 <2>
+spec:
+  provider: aws                           <3>
+  parameters:                             <4>
+ifdef::aws-secrets-manager[]
+    objects: |
+      - objectName: "testSecret"
+        objectType: "secretsmanager"
+endif::aws-secrets-manager[]
+ifdef::aws-systems-manager-parameter-store[]
+    objects: |
+      - objectName: "testSecret"
+        objectType: "ssmparameter"
+endif::aws-systems-manager-parameter-store[]
+----
+<1> Specify the name for the secret provider class.
+<2> Specify the namespace for the secret provider class.
+<3> Specify the provider as `aws`.
+<4> Specify the provider-specific configuration parameters.
+
+.. Create the `SecretProviderClass` object by running the following command:
++
+[source,terminal]
+----
+$ oc create -f secret-provider-class-aws.yaml
+----
+
+. Create a deployment to use this secret provider class:
+
+.. Create a YAML file that defines the `Deployment` object:
++
+.Example `deployment.yaml`
+[source,yaml]
+----
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: my-aws-deployment                              <1>
+  namespace: my-namespace                              <2>
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: my-storage
+  template:
+    metadata:
+      labels:
+        app: my-storage
+    spec:
+      containers:
+      - name: busybox
+        image: k8s.gcr.io/e2e-test-images/busybox:1.29
+        command:
+          - "/bin/sleep"
+          - "10000"
+        volumeMounts:
+        - name: secrets-store-inline
+          mountPath: "/mnt/secrets-store"
+          readOnly: true
+      volumes:
+        - name: secrets-store-inline
+          csi:
+            driver: secrets-store.csi.k8s.io
+            readOnly: true
+            volumeAttributes:
+              secretProviderClass: "my-aws-provider" <3>
+----
+<1> Specify the name for the deployment.
+<2> Specify the namespace for the deployment. This must be the same namespace as the secret provider class.
+<3> Specify the name of the secret provider class.
+
+.. Create the `Deployment` object by running the following command:
++
+[source,terminal]
+----
+$ oc create -f deployment.yaml
+----
+
+.Verification
+
+* Verify that you can access the secrets from {secrets-store-provider} in the pod volume mount:
+
+.. List the secrets in the pod mount:
++
+[source,terminal]
+----
+$ oc exec busybox-<hash> -n my-namespace -- ls /mnt/secrets-store/
+----
++
+.Example output
+[source,terminal]
+----
+testSecret
+----
+
+.. View a secret in the pod mount:
++
+[source,terminal]
+----
+$ oc exec busybox-<hash> -n my-namespace -- cat /mnt/secrets-store/testSecret
+----
++
+.Example output
+[source,terminal]
+----
+{"username":"my-user", "password":"my-password"}
```

Comment:
The secret value stems from the case's command `aws --region "$REGION" secretsmanager  create-secret --name xxiaSecret --secret-string '{"username":"memeuser", "password":"hunter2"}' --description "xxia test purpose"`. It can be any secret value. Will users be confuse where `{"username":"my-user", "password":"my-password"}` stems from? Is it better to be general like below?
```suggestion
<secret value>
```

Reply:
It is supposed to just be an example output, but I can definitely put just a replaceable value there if that would be less confusing